opensearch_py_ml/ml_commons/IAMRoleHelper.py

```diff
@@ -0,0 +1,301 @@
+            print(f"ARN: {role['Arn']}")
+            print(f"Creation Date: {role['CreateDate']}")
+            print("Assume Role Policy Document:")
+            print(
+                json.dumps(role["AssumeRolePolicyDocument"], indent=4, sort_keys=True)
+            )
+
+            # List and print all inline policies attached to the role
+            list_role_policies_response = iam.list_role_policies(RoleName=role_name)
+
+            for policy_name in list_role_policies_response["PolicyNames"]:
+                get_role_policy_response = iam.get_role_policy(
+                    RoleName=role_name, PolicyName=policy_name
+                )
+                print(f"Role Policy Name: {get_role_policy_response['PolicyName']}")
+                print("Role Policy Document:")
+                print(
+                    json.dumps(
+                        get_role_policy_response["PolicyDocument"],
+                        indent=4,
+                        sort_keys=True,
+                    )
+                )
+
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "NoSuchEntity":
+                print(f"Role {role_name} does not exist.")
+            else:
+                print(f"An error occurred: {e}")
+
+    def get_user_arn(self, username):
+        """
+        Retrieve the ARN of an IAM user.
+
+        :param username: Name of the IAM user.
+        :return: ARN of the user or None if not found.
+        """
+        if not username:
+            return None
+        iam_client = boto3.client("iam")
+
+        try:
+            response = iam_client.get_user(UserName=username)
+            user_arn = response["User"]["Arn"]
+            return user_arn
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "NoSuchEntity":
+                print(f"IAM user '{username}' not found.")
+                return None
+            else:
+                print(f"An error occurred: {e}")
+                return None
+
+    def assume_role(self, role_arn, role_session_name="your_session_name"):
+        """
+        Assume an IAM role and obtain temporary security credentials.
+
+        :param role_arn: ARN of the IAM role to assume.
+        :param role_session_name: Identifier for the assumed role session.
+        :return: Temporary security credentials or None if the operation fails.
+        """
+        sts_client = boto3.client("sts")
+
+        try:
+            assumed_role_object = sts_client.assume_role(
+                RoleArn=role_arn,
+                RoleSessionName=role_session_name,
+            )
+
+            # Extract temporary credentials from the assumed role
+            temp_credentials = assumed_role_object["Credentials"]
+
+            return temp_credentials
+        except ClientError as e:
+            print(f"Error assuming role: {e}")
+            return None
+
+    def map_iam_role_to_backend_role(self, iam_role_arn):
```

Comment:
I don't think this method should be part of this class. This is mapping a role to the opensearch backend role. Let's put it somewhere else.